donation/views.py

```diff
@@ -16,15 +17,26 @@ def default(self, obj):
 
 # Create your views here.
 def donation_create(request):
+    if request.user.is_anonymous:
+        form= CreateNewDonation()
+    else:
+        form = CreateNewDonation(initial={'ong': request.user.ong})
     if request.method == "POST":
         form = CreateNewDonation(request.POST)
         if form.is_valid():
+            if not request.user.is_anonymous:
+                ong=request.user.ong
+                donation=form.save()
+                donation.ong=ong
+                donation.save()
             form.save()
+            return redirect("/donation/list")
         else:
-            print(form.errors) 
+            print(form)
+            messages.error(request, 'Formulario con errores')
 
-    form = CreateNewDonation()
-    return render(request, 'donation_form.html', {"form": form, "title": "Crear Donación"})
+    
+    return render(request, 'donation/create.html', {'object_name': 'donate', "form": form, "button_text": "Registrar donación"})
```

Comment:
Maybe you don't need to pass through the context the button_text and just include inside the template. This is just a suggestion.